File: api/clients/src/main/java/com/foodagram/clients/notification/NotificationDto.java
package com.foodagram.clients.notification;

import com.foodagram.clients.notification.dto.NotificationStatus;
import com.foodagram.clients.notification.dto.NotificationType;
import lombok.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationDto {
    private UUID id;
    private UUID userId;
    private String title;
    private String content;
    private NotificationStatus notificationStatus;
    private NotificationType notificationType;
    private String targetUrl;
    private UUID senderId;
    private LocalDateTime createdDate;
    private LocalDateTime updatedDate;
    private String mediaUrl;
    private Boolean forAdmins;
}

File: fooadgram-app/models/notification/NotificationDto.ts
import {NotificationStatus, NotificationType} from "@/models/notification/NotificationCreateDto";

export interface NotificationDto {
    id: string; // UUID
    userId: string;
    title: string;
    content: string;
    notificationStatus: NotificationStatus;
    notificationType: NotificationType;
    targetUrl?: string;
    senderId?: string;
    createdDate: string; // ISO string (e.g., "2025-04-26T14:00:00Z")
    updatedDate: string;
    mediaUrl: string;
}
